modules/nf-core/gmmdemux/main.nf

```diff
@@ -0,0 +1,71 @@
+
+process GMMDEMUX {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gmm-demux:0.2.2.3--pyh7cba7a3_0':
+        'biocontainers/gmm-demux:0.2.2.3--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(meta), path(cell_hashing_barcodes,stageAs: "hto_files/*"), path(cell_hashing_matrix,stageAs: "hto_files/*"),path(cell_hashing_features,stageAs: "hto_files/*"),val(hto_names)
+    val csv
+    val num_cells
+    val output_dir
+    val full_report
+    val simplified_report
+    val examine
+    val ambigous
+    val extract
+    val skip
+
+    output:
+    tuple val(meta), path('test/barcodes.tsv.gz'), emit: barcodes
+    tuple val(meta), path('test/*.mtx.gz'), emit: matrix
+    tuple val(meta), path('test/features.tsv.gz'), emit: features
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    //since this tool has many optional inputs that can be passed in, we need to check if they are null or not
+    // in order to produce or not certain reports or outputs
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def n_cells = num_cells ? "--summary $num_cells" : ""
+    def output_path = output_dir ? "-o $prefix" : ""
+    def full_rep = full_report ? "-f $prefix" : ""
+    def simplified_rep = simplified_report ? "--simplified $simplified_report" : ""
+    def examine = examine ? "--examine $examine" : ""
+    def extract = extract ? "--extract $extract" : ""
+    def skip = skip ? "--skip $skip" : ""
+    def type = csv ? "-c" : ""
+    def ambigous = examine ? "--ambigous $ambigous " : ""
+    def VERSION = '0.2.2.3' // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
+    """
+    GMM-demux $type hto_files $hto_names $output_path $full_rep $n_cells $simplified_rep $examine $ambigous $extract $skip $args
```

Comment:
```suggestion
    GMM-demux \\
       $type \\
       $hto_files \\
       $hto_names <...> \\
       $skip \\
       $args
```

Please split over multiple lines, e.g. the abbove. Also maybe indicate hto_files is a directory as you stage them in `hto_files/`

Reply:
I had this structure, but sadly, the arguments cannot be passed with a new line as required by the style guides from nf-core, since they break the execution of the tool and generate an error for unknown parameters. I'll reproduce the error and show you

Reply:
Here is a reproduction of the error when splitting in multiple lines
```
 Nextflow stdout:                                                                                                                                                      │
│                                                                                                                                                                         │
│   ERROR ~ Error executing process > 'GMMDEMUX (test)'                                                                                                                   │
│                                                                                                                                                                         │
│   Caused by:                                                                                                                                                            │
│     Process `GMMDEMUX (test)` terminated with an error exit status (2)                                                                                                  │
│                                                                                                                                                                         │
│   Command executed:                                                                                                                                                     │
│                                                                                                                                                                         │
│     GMM-demux  \                                                                                                                                                        │
│         hto_files \                                                                                                                                                     │
│         MS-11,MS-12 \                                                                                                                                                   │
│          \                                                                                                                                                              │
│          \                                                                                                                                                              │
│          \                                                                                                                                                              │
│          \                                                                                                                                                              │
│          \                                                                                                                                                              │
│          \                                                                                                                                                              │
│     cat <<-END_VERSIONS > versions.yml                                                                                                                                  │
│     "GMMDEMUX":                                                                                                                                                         │
│         GMM-Demux: 0.2.2.3                                                                                                                                              │
│     END_VERSIONS                                                                                                                                                        │
│                                                                                                                                                                         │
│   Command exit status:                                                                                                                                                  │
│     2                                                                                                                                                                   │
│                                                                                                                                                                         │
│   Command output:                                                                                                                                                       │
│     ==============================GMM-Demux Initialization==============================                                                                                │
│     Confidence threshold: 0.8                                                                                                                                           │
│                                                                                                                                                                         │
│   Command error:                                                                                                                                                        │
│     usage: GMM-demux [-h] [-k SKIP] [-x EXTRACT] [-o OUTPUT] [-f FULL] [-c]                                                                                             │
│                      [-t THRESHOLD] [-s SIMPLIFIED] [-u SUMMARY] [-r REPORT]                                                                                            │
│                      [-e EXAMINE] [-a AMBIGUOUS] [-rs RANDOM_SEED]                                                                                                      │
│                      [input_path ...] [hto_array ...] input_path hto_array                                                                                              │
│     ==============================GMM-Demux Initialization==============================                                                                                │
│     Confidence threshold: 0.8                                                                                                                                           │
│     GMM-demux: error: the following arguments are required: hto_array                                                                                                   │
│                                                                                                                                                                         │
│   Work dir:                                                                                             
```

Reply:
I'm not sure  I follow, what is the `hto_array` here? The MS11-MS-12 thing? The line breaks are a bash thing so should work...(?), as it just knows to stick everything together into a single line when evaluating the commnad

Reply:
`hto_array` contains a string with the sample barcoding tags, separated by commas, present in the hto data.
You're right with the bash thing, perhaps is it related to the static variable `hto_files`?